### PR TITLE
Splitting TestUTestMacro.cpp

### PR DESCRIPTION
**Pull request by arstrube**
This isn't particularly elegant; I was trying to keep new files limited to one and avoid duplication of code.

1. Only the first commit is of interest; the rest will be taken care of by basvodde/long-long.

2. Better ideas?

3. How to divide up the file more sensibly? Or can we shrink the code (a lot of assembler instructions will be generated for doing 64 bit math on a 16 bit platform)?

### CODE REVIEW

File: include/CppUTest/TestTestingFixture.h [hunk 1]
```diff
@@ -121,11 +123,40 @@ class TestTestingFixture
...
+        return lineOfCodeExecutedAfterCheck();
+    }
 
     TestRegistry* registry_;
     ExecFunctionTestShell* genTest_;
     StringBufferTestOutput* output_;
     TestResult * result_;
+    
+private:
+    static bool lineOfCodeExecutedAfterCheck(const bool* const value = 0)
```
Comment (arstrube) [inline include/CppUTest/TestTestingFixture.h]: This is a bit of a hack, but it works without having to introduce a new .cpp file just for the static var to live in.

File: tests/TestUTestMacro2.cpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        TestTestingFixture::setLineOfCodeExecutedAfterCheck(false);
+    }
+
+    void runTestWithMethod(void(*method)())
+    {
+        fixture.setTestFunction(method);
+        fixture.runAllTests();
+    }
+};
+
```
Comment (arstrube) [inline tests/TestUTestMacro2.cpp]: I just pulled out the long  long tests; this will work, but isn't particularly smart ;-)

**Comment by basvodde** [comment]
Alternative split at #953

**Comment by arstrube** [comment]
@basvodde fixing some errors this end - not sure how this can be:
```
In file included from ../tests/CppUTestExt/GTest1Test.cpp:83:
../include/CppUTest/TestTestingFixture.h:130:13: error: use of undeclared 
identifier 'FAIL_LOCATION'
FAIL_LOCATION(StringFromFormat("Expected one test failure, b...
^
../include/CppUTest/TestTestingFixture.h:135:13: error: use of undeclared 
identifier 'FAIL_LOCATION'
FAIL_LOCATION("The test should jump/throw on failure and no...
^
```
seeing that TestHarness.h is included before TestTestingFixture.h, and includes UTestMacro.h in turn? (of course it would be cleaner to implement this method in TestTestingFixture.cpp and #include TestHarness.h there, but that is an entirely different matter). The macro should be known? Why does it only cause problems in GTest1Test.cpp?

**Pull request closed without merging**